packages/notify-client/src/controllers/engine.ts

```diff
@@ -336,6 +337,71 @@ export class NotifyEngine extends INotifyEngine {
     return true;
   };
 
+  public getNotificationHistory: INotifyEngine["getNotificationHistory"] =
+    async ({ topic, limit, startingAfter }) => {
+      this.isInitialized();
+
+      const subscription = this.client.subscriptions.get(topic);
+
+      const identityKey = encodeEd25519Key(
+        await this.client.identityKeys.getIdentity({
+          account: subscription.account,
+        })
+      );
+
+      const issuedAt = Math.round(Date.now() / 1000);
+      const expiry =
+        issuedAt + ENGINE_RPC_OPTS["wc_notifyGetNotifications"].res.ttl;
+
+      const cachedKey = this.getCachedDappKey(subscription);
+      const dappUrl = getDappUrl(subscription.metadata.appDomain);
+      const { dappIdentityKey } = cachedKey
+        ? { dappIdentityKey: cachedKey }
+        : await this.resolveKeys(dappUrl);
+
+      const getNotificationsClaims: NotifyClientTypes.GetNotificationsJwtClaims =
+        {
+          act: "notify_get_notifications",
```

Comment:
```suggestion
          action: "notify_get_notifications",
```

I don't know if this is the right name but I wanted to point out that this field and other fields in this object might be better to have more readable field names. When I read the code, I didn't understand if these values were correctly set without knowing the codebase as a reviewer. I believe we should prever readable field names too.

Reply:
The fields unfortunately have to be compliant with the spec linked above. See the "Authentication" page for details. The reason for the short, 3 letter names is to cut down redundant data on transport, since this all gets encoded into a JWT.

Reply:
As mentioned below in my other comment this is to be spec compliant: https://github.com/WalletConnect/walletconnect-specs/pull/183/files#diff-02dc8b9dd052b54a577b55ecf0d2b3ad2d58f1df0a0be7ba388fa721ffcbc48eR137

Would have to be agreed at the x-platform level to use longer form but I believe this is the convention for JWT claims